lib/rules/one-var.js

```diff
@@ -272,6 +276,87 @@ module.exports = {
+            const declaration = declarations[0];
+
+            return fixer => {
+                const type = sourceCode.getTokenBefore(declaration);
+                const prevSemi = sourceCode.getTokenBefore(type);
+                const prevType = sourceCode.getTokenBefore(type, {
+                    filter(token) {
+                        return ~["const", "var", "let"].indexOf(token.value) && token.type === "Keyword";
+                    }
+                });
+                const res = [];
+
+                if (sourceCode.getText(prevType) === sourceCode.getText(type)) {
+                    res.push(fixer.replaceText(prevSemi, ","));
+                    res.push(fixer.replaceText(type, ""));
+                }
+
+                return res;
+            };
+        }
+
+        /**
+         * Fixer to split a VariableDeclaration into individual declarations
+         * @param   {VariableDeclaration}   declaration The `VariableDeclaration` to split
+         * @param   {?Function}             filter      Function to filter the declarations
+         * @returns {Function}                          The fixer function
+         */
+        function splitDeclarations(declaration) {
+            return fixer => declaration.declarations.map(declarator => {
+                const comma = sourceCode.getTokenAfter(declarator);
+                const afterComma = sourceCode.getTokenAfter(comma, { includeComments: true });
+
+                if (comma.value === ",") {
+                    let spaceBefore = " ";
+                    let spaceAfter = "";
+
+                    if (afterComma.range[0] === comma.range[1]) {
+
+                        /*
+                         * `var x,y`
+                         * comma ^^ afterComma
+                         */
+                        spaceAfter = " ";
+                        return fixer.replaceText(comma, `;${spaceBefore}${declaration.kind}${spaceAfter}`);
+                    }
+                    if (afterComma.loc.start.line > comma.loc.end.line || afterComma.type === "Line" || afterComma.type === "Block") {
+
+                        /*
+                         * `var x,
+                         * comma ^
+                         *      y`
+                         *      ^ afterComma
+                         */
+                        spaceBefore = "\n";
+                        let lastComments = afterComma;
+                        const comments = [];
+
+                        while (lastComments.type === "Line" || lastComments.type === "Block") {
+                            comments.push(lastComments);
+                            lastComments = sourceCode.getTokenAfter(lastComments, { includeComments: true });
+                        }
+                        let replacedText = "";
+
+                        for (const comment of comments) {
+                            replacedText += `${sourceCode.getText(comment)}\n`;
```

Comment:
Instead of joining all the comments with newlines, would it be possible to do something like this instead?

```js
let lastComment = afterComma;

while (lastComment.type === "Line" || lastComment.type === "Block") {
    lastComment = sourceCode.getTokenAfter(lastComment, { includeComments: true });
}

return fixer.replaceTextRange(
    [comma.range[0], lastComment.range[0]],
    `;\n${sourceCode.text.slice(comma.range[1], lastComment.range[0])}\n${declaration.kind} `
);
```